[Topic pages] Publish topic pages

jankrcal opened the pull request:
This PR makes all the changes necessary for publishing topic pages:
- reworks tags.html to allow printing topics that list a given page
- rework navigation .html to allow listing topics
- hide tags on preview boxes on the old tags page (to make tests happy; it's disputable if it ever was a good idea to list tags there)
- add simple css rules for topic tiles for the main page.

jankrcal commented:
@mukrop This is the web-core PR I mentioned over slack.

Please do not merge, I am waiting for one more PR on the web-cz side to land.

mukrop reviewed (approved):
As discussed, I find it a great work (and code-wise also very diligent). A couple of comments below, none of which is blocking.

Some comments are in code, others below.

* Just to make sure of the conceptual decision: This change means each new object will need to be _manually_ added to a topic+subtopic when added online?
(I don't think it's bad, it may nicely force the authors to think about where it fits – I'm just making sure we are consciously taking this conceptual change).
* Numbers at the top of the topic page: I'm proud they link the source though the visual positioning of the "external" icon seems a bit odd, often in the middle of empty space. I understand you wanted to be minimalistic (without saying "Source" in each). I'm unsure what would be a better solution, though... My thoughts: Maybe the "i" (info) symbol instead of "external"? Maybe have the number be the link with no symbol? Maybe keep the external symbol but position it after the end of the text under the number?
* The topic navbar has a bit odd animation when disappearing – this is especially prominent on desktop when scrolling up a bit faster. As it disappears gradually, you can see the navbar disappearing somewhat "in the middle of the page". Maybe just disappear faster?
* Clicking the link "More materials" below the expandable card list is sometimes tricky to click. As only the text is the link (and not an underlying block element), if you position the cursor in the cavity of "e" for example, the link does not work.
* On the topic page, the "In a nutshell" section (_V kostce_) is not in the navbar, yet uses the same style as the navbar. Not tragic, just confused me a bit. Maybe having the link to "V kostce" instead of the current link of "Uvod" and adding a link to the topic name (above subtopics in navbar) pointing that to the top of the page?
* I see the topic pages are almost entirely structured in the header. I understand the reason, though having the main lead description on the very bottom of the file then seems rather counter-intuitive when editing the file (everything else follows in logical progression as you see it on the page). Wouldn't it be worth moving this part to a structured field as well to ease the orientation for editors? (Yes, the page would then have no body but I don't think it would be a problem.)

* (Probably unrelated visual issue) The recherche tile has text snippet in the tile. This is unlike any other type – wouldn't it be better to use a placeholder image for this type (as is the case for the podcast, for example) and drop the text? 
* (Again, unrelated visual idea) Most tile types explicitly say the type but infographics don't. If we moved the type to the lower part of the tile, infographics could also have them bringing a bit more visual consistency.

jankrcal commented on _includes/header.html:
```diff
@@ -53,7 +53,7 @@
 <link rel="preconnect" href="https://fonts.googleapis.com">
 <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
-<link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;700&family=Source+Sans+Pro:ital,wght@0,400;0,700;1,400&display=swap" rel="stylesheet">
+<link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;700&family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400&display=swap" rel="stylesheet">
```
According to Kristi, it wouldn't work.

I'd rather strive for much wider use of semi-bold to make it worth it.

Btw, it's one more typeface, we already had 3 (regular, bold, italic), so it should "only" be 33 % more.

jankrcal commented on _includes/tags.html:
```diff
@@ -1,5 +1,35 @@
+        {%- endfor %}
+        {%- if subtopic_id %}
+            {%- if include.highlighted == topic.slug %}
+            <a href="/" role="button" class="btn btn-tag btn-sm btn-tag-highlighted">
```
I don't want to fix the pre-existing code in this CL. You're right this is weird and even unreachable code (as this is only used from tag.html which is not used together with new topic pages). Removed this case.

jankrcal replied:
I'm hesitant to fix preexisting code in this rather large CL. You're right this branch is not needed in the new code (as this is only used from tag.html which is not used with new topic pages), removed it.

jankrcal commented on _includes/preview-blocks.html:
```diff
@@ -12,7 +12,7 @@
                 <a href="{{ item.url }}" class="preview-card card">
             {% endif %}
-            {% include preview-block-without-link.html item=item %}
+            {% include preview-block-without-link.html item=item no_include_tags=include.no_include_tags %}
```
Good idea to get rid of it completely. Filed #192.

jankrcal commented on _layouts/tag.html:
```diff
@@ -46,7 +46,7 @@ <h2>{{ collection.title }} <small class="text-secondary">({{ items_size }})</sma
               {% when 'datasets' %}
               <p class="lead">{{ site.data.lang.tag.all-datasets }}</p>
             {% endcase %}
-            {% include preview-blocks.html blocks=items %}
```
Opened #193 to cover this.

Essentially, I am keeping backwards compatibility. I wasn't necessarily planning to migrate sk and en that soon (as they don't have too much content so the current design kind of works) but we can maybe use the new design appropriately (and cheaply) to make the migration quicker.

jankrcal commented:
> As discussed, I find it a great work (and code-wise also very diligent). A couple of comments below, none of which is blocking.
> 
> Some comments are in code, others below.
> 
> * Just to make sure of the conceptual decision: This change means each new object will need to be _manually_ added to a topic+subtopic when added online?
>   (I don't think it's bad, it may nicely force the authors to think about where it fits – I'm just making sure we are consciously taking this conceptual change).
Yes, this is the decision: there's now also presubmit check that doesn't let you merge it if you forget to include it in any topic+subtopic (making sure there's no <div class="build-error"> in the result)

> * Numbers at the top of the topic page: I'm proud they link the source though the visual positioning of the "external" icon seems a bit odd, often in the middle of empty space. I understand you wanted to be minimalistic (without saying "Source" in each). I'm unsure what would be a better solution, though... My thoughts: Maybe the "i" (info) symbol instead of "external"? Maybe have the number be the link with no symbol? Maybe keep the external symbol but position it after the end of the text under the number?
We've spent quite a bit time exploring alternatives, so I'd stick to what we have now. I'll move it to a bug with design proposals and consult with Kristi, anyway.

> * The topic navbar has a bit odd animation when disappearing – this is especially prominent on desktop when scrolling up a bit faster. As it disappears gradually, you can see the navbar disappearing somewhat "in the middle of the page". Maybe just disappear faster?
Good feedback, I've removed the animation for disappearing (only the small title fades out which is enough for the feeling of smoothness and kept the existing animations for appearing).

> * Clicking the link "More materials" below the expandable card list is sometimes tricky to click. As only the text is the link (and not an underlying block element), if you position the cursor in the cavity of "e" for example, the link does not work.
This was caused by a different problem, I think - fixed.

> * On the topic page, the "In a nutshell" section (_V kostce_) is not in the navbar, yet uses the same style as the navbar. Not tragic, just confused me a bit. Maybe having the link to "V kostce" instead of the current link of "Uvod" and adding a link to the topic name (above subtopics in navbar) pointing that to the top of the page?
Moved to a new issue. We've also spent quite some time exploring so not sure we'll change that.

> * I see the topic pages are almost entirely structured in the header. I understand the reason, though having the main lead description on the very bottom of the file then seems rather counter-intuitive when editing the file (everything else follows in logical progression as you see it on the page). Wouldn't it be worth moving this part to a structured field as well to ease the orientation for editors? (Yes, the page would then have no body but I don't think it would be a problem.)
Good point, done!

> * (Probably unrelated visual issue) The recherche tile has text snippet in the tile. This is unlike any other type – wouldn't it be better to use a placeholder image for this type (as is the case for the podcast, for example) and drop the text?
Long-term, I think it would be better to have a cover images (like explainers). Does not have the priority, though.

> * (Again, unrelated visual idea) Most tile types explicitly say the type but infographics don't. If we moved the type to the lower part of the tile, infographics could also have them bringing a bit more visual consistency.
I am aware of the asymmetry but I don't want to increase consistency at the cost of more visual clutter.
